### PR TITLE
The Big Merge

### LINKED ISSUE #6: UPDATE: Update the documentation to reflect the actual pipeline
### Description of feature

The documentation is out of data and needs updating to reflect the actual pipeline arguments required

### PR DESCRIPTION
**Pull request by DLBPointon**
Closes #6 

Release ready pipeline!

### CODE REVIEW

**Comment by priyanka-surana** [comment]
@DLBPointon Do you want to fix the CI tests in this pipeline too?

**Comment by DLBPointon** [comment]
> @DLBPointon Do you want to fix the CI tests in this pipeline too? 

I will be. I'll ask Guoying about it tomorrow, as it's all tower stuff, we are in an ascc Hackathon at the minute.

**Comment by DLBPointon** [comment]
> The tower agent was down two days ago. I re-run the job and submitted it to Tower successfully.
> 
> But the pipeline failed to run. https://tower.nf/orgs/sanger-tol/workspaces/github-ci/watch/4Mp053V8l2TU1H
> 
> ```
> 
> ERROR ~ ERROR: Validation of pipeline parameters failed!
> 
>  -- Check 'nf-4Mp053V8l2TU1H.log' file for details
> ERROR ~ * --input: string [/lustre/scratch123/tol/resources/treeval/treeval-testdata/TreeValSmallData/Oscheius_DF5033/assembly/draft/DF5033.hifiasm.noTelos.20211120/DF5033.noTelos.hifiasm.purged.noCont.noMito.fasta] does not match pattern ^\S+\.fa$ (/lustre/scratch123/tol/resources/treeval/treeval-testdata/TreeValSmallData/Oscheius_DF5033/assembly/draft/DF5033.hifiasm.noTelos.20211120/DF5033.noTelos.hifiasm.purged.noCont.noMito.fasta)
> 
>  -- Check 'nf-4Mp053V8l2TU1H.log' file for details
> 
> Exiting!
> ```

Hi Guoying, 
Thanks for having a look, I've fixed the issue in: https://github.com/sanger-tol/curationpretext/pull/13 
The Schema just needed updating, I also changed the outdir value other wise it would constantly come back as null in tests.

**Comment by gq1** [comment]
> Hi Guoying, Thanks for having a look, I've fixed the issue in: #13 The Schema just needed updating, I also changed the outdir value other wise it would constantly come back as null in tests.

Can you re-trigger the Github action run to test?

**Comment by DLBPointon** [comment]
Morning @priyanka-surana, @gq1
Looks like all tests including Tower have been completed successfully.

**Comment by DLBPointon** [comment]
Hi @priyanka-surana,
Can this now be approved?

**Comment by DLBPointon** [comment]
> Hi @priyanka-surana, Can this now be approved?

Never mind this should wait on #14

File: conf/base.config [hunk 1]
```diff
@@ -45,7 +45,7 @@ process {
     }
 
     withName: BWAMEM2_INDEX {
-        cpus    = {}        
+        memory  = { check_max( 100.GB  * task.attempt, 'memory' ) }
```
Comment (priyanka-surana) [inline conf/base.config:48]: Does `BWAMEM2` really need 100GB even for small assemblies? Similarly for `CRAM_FILTER_ALIGN_BWAMEM2_FIXMATE_SORT`.

**Review by priyanka-surana — CHANGES_REQUESTED** [review]
I tried the `test` profile and it fails for me.
```
nextflow run . -profile test,singularity --outdir results -ansi-log false
ERROR ~ Unable to read script: '/lustre/scratch123/tol/teams/tolit/users/ps22/pipelines/sanger-tol-curationpretext_dev/dev/./workflows/curationpretext_allf.nf' -- cause: /home/runner/work/curationpretext/curationpretext/TreeValTinyData/genomic_data/pacbio
```
I can see the test profile is set for GitHub, which profile should I be using to test the subset and full datasets.

**Comment by gq1** [comment]
I tried the `test` profile on the farm as well and got the same error message.

Can we have the instruction as @muffato did to TreeVal to have some instructions to download the test file from S3 and replace all the github path to the local path?

> 
> I can see the test profile is set for GitHub, which profile should I be using to test the subset and full datasets? Thanks.

I also tried the `test_full` profile directly on the farm, it works fine. This profile already passed on the farm via Github action and Tower.

File: conf/base.config (hunk 1 above)
Comment (DLBPointon) [reply conf/base.config:48]: Decreased this to 25GB, your right it wouldn't need 100.
Done

**Comment by DLBPointon** [comment]
Thanks both @gq1 and @priyanka-surana.
I've submitted fixes, they are on #14. usage now contains information on amending the test.config.

**Comment by DLBPointon** [comment]
#14 is now merged

**Pull request merged**